libs/openFrameworks/types/ofColor.cpp

```diff
@@ -426,7 +426,7 @@ void ofColor_<PixelType>::setHsb(float hue, float saturation, float brightness,
 	} else {
 		float hueSix = hue * 6.f / limit();
 		float saturationNorm = saturation / limit();
-		int hueSixCategory = (int) floorf(hueSix);
+		int hueSixCategory = std::floor( (int)hueSix );
```

Comment:
To match the previous functionality, the (int) cast should be outside of the floor function.
`int hueSixCategory = (int)std::floor( hueSix );`

Reply:
Thank you Nick for spotting this error.

Reply:
code updated